pkg/kinflate/app/application.go

```diff
@@ -61,8 +63,25 @@ func New(loader loader.Loader) (Application, error) {
 
 // Resources computes and returns the resources from the manifest.
 func (a *applicationImpl) Resources() (resource.ResourceCollection, error) {
```

Comment:
Please add a comment explaining name hashing and name ref are only updated at the very end.